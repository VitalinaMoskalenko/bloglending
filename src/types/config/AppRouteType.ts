export enum AppRouteType {
  home = "/",
  postDetail = "/postDetail",
  postDetailsWithId = "/postDetail/:id",
}
