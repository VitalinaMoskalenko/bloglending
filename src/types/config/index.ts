export * from "./FontSizeType";
export * from "./ColorType";
export * from "./AppLocalsType";
export * from "./BreakpointType";
export * from "./AppRouteType";
