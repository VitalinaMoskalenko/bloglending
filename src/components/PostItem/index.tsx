import React from "react";
import styled from "styled-components";
import { BreakpointType } from "../../types";
import { H3 } from "../Headings";
import { Body } from "../Paragraphs";

const Container = styled.li`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  height: fit-content;
  padding: 18px;
  border-radius: 8px;
  margin-bottom: 10px;
  list-style-type: none;
  cursor: pointer;

  @media (max-width: ${BreakpointType.xs}px) {
    height: fit-content;
    align-items: flex-start;
    flex-direction: column;
  }
  :hover {
    opacity: 0.5;
  }
`;

const PostContainer = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
  align-items: center;
`;

const Avatar = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 20px;
`;

const DesktopType = styled(Body)`
  flex: 1;
  @media (max-width: ${BreakpointType.xs}px) {
    display: none;
  }
`;

const MobileType = styled(Body)`
  display: none;

  @media (max-width: ${BreakpointType.xs}px) {
    display: flex;
  }
`;

const Info = styled.div`
  margin-left: 20px;

  @media (max-width: ${BreakpointType.xs}px) {
    display: flex;
    flex-direction: column;
  }
`;

const Description = styled(Body)`
  flex: 1;
  width: 150px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  @media (max-width: ${BreakpointType.xs}px) {
    margin-top: 12px;
    width: 100%;
  }
`;

type PropsType = {
  name: string;
  description: string;
  onClick: () => void;
};

const PostItem = ({ name, description, onClick }: PropsType) => {
  return (
    <Container onClick={onClick}>
      <PostContainer>
        <Info>
          <H3>{name}</H3>
        </Info>
      </PostContainer>
      <Description>{description}</Description>
    </Container>
  );
};

export default PostItem;
