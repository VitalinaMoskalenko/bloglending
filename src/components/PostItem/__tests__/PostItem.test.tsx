import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import PostItem from "../";
import { theme } from "../../../config";

it("renders PostItem correctly", () => {
  const postItem = renderer
    .create(
      <ThemeProvider theme={theme}>
        <PostItem name="name" description="description" onClick={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(postItem).toMatchSnapshot();
});
