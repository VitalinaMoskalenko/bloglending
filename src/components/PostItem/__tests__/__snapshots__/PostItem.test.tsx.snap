// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PostItem correctly 1`] = `
<li
  className="sc-iCoGMd hNhGTR"
  onClick={[Function]}
>
  <div
    className="sc-fujyAs hMTAzb"
  >
    <div
      className="sc-iqAclL bsILWn"
    >
      <h1
        className="sc-dlnjwi czuNcg"
      >
        name
      </h1>
    </div>
  </div>
  <p
    className="sc-jSFjdj sc-crzoAE bQuzKy fqRwMk"
  >
    description
  </p>
</li>
`;
